test/helpers/policygen/models.go

```diff
@@ -702,6 +702,9 @@ func (t *TestSpec) InvalidNetworkPolicyApply() (*cnpv2.CiliumNetworkPolicy, erro
 		if cnp != nil {
 			return true
 		}
+		if len(cnp.Status.Nodes) == 0 {
```

Comment:
This will never get hit, because `cnp != nil` will always evaluate to `true` and thus `body` will always return `true` before this logic is hit - shouldn't this check be within the `if cnp != nil` block?